basheditor-plugin/src/main/java-eclipse/de/jcup/basheditor/BashEditor.java

```diff
@@ -774,27 +774,32 @@ protected void performSave(boolean overwrite, IProgressMonitor progressMonitor)
 				
 				// refreshing does not seem to work:
 				//resolveResourceAsIFile().refreshLocal(IResource.DEPTH_INFINITE, progressMonitor);
+				ISourceViewer sourceViewer = getSourceViewer();
+				StyledText textWidget = sourceViewer.getTextWidget();
+				int lastCaretPos = textWidget.getCaretOffset();
 				
 				// reload changes performed on disk by external reformatter:
 				String external_tool_result = readFile(bashFile.getAbsolutePath(), StandardCharsets.UTF_8);
-
-				//int caretPositionBeforeReload = lastCaretPosition;
-				/*getDocument().set(external_tool_result);
-				BashEditorUtil.logInfo("Moving the caret at: " + caretPositionBeforeReload);
-				ISourceViewer sourceViewer = getSourceViewer();
-				StyledText textWidget = sourceViewer.getTextWidget();
-				textWidget.setCaretOffset(caretPositionBeforeReload);
-				textWidget.showCaret();*/
-
-				// replace text in our text widget:
-				ISourceViewer sourceViewer = getSourceViewer();
-				StyledText textWidget = sourceViewer.getTextWidget();
-				int loc = textWidget.getCaretOffset();
-				textWidget.replaceTextRange(0, textWidget.getCharCount(), external_tool_result);
 				
+				// SOLUTION #1: change the Document object directly
+				getDocument().set(external_tool_result);
+				
+				// SOLUTION #2: use widget replaceText
+				//textWidget.replaceTextRange(0, textWidget.getCharCount(), external_tool_result);
+				
+				// PROBLEM: WHAT'S THE BEST APPROACH? BOTH REQUIRE RESTORING THE CARET POSITION ANYWAY
+
 				// hackish way to try to restore the caret to the position it had BEFORE content was reformatted
-				textWidget.setCaretOffset(loc);
+				textWidget.setCaretOffset(lastCaretPos);
+				BashEditorUtil.logInfo("Moved the caret at: " + textWidget.getCaretOffset());
 				textWidget.showSelection();
+
+				// PROBLEM: IN THIS WAY THE DOCUMENT IS MARKED AS "DIRTY" BY ECLIPSE AND THUS
+				// AN ASTERISK WILL APPEAR IN THAT DOCUMENT TAB AND ECLIPSE WILL ASK TO RELOAD IT
+				// SINCE IT WAS CHANGED ON DISK!!!
+
+				// hackish way to remove the "dirty" status from the document: re-save with overwrite=true
+				super.performSave(true, progressMonitor);
```

Comment:
I think this is okay . Maybe https://wiki.eclipse.org/FAQ_When_should_I_use_refreshLocal%3F would be another option which could save much code here.